src/vulkan/buffer_descriptor.cc

```diff
@@ -39,17 +42,24 @@ Result BufferDescriptor::CreateResourceIfNeeded() {
         "only when |transfer_buffers| is empty");
   }
 
-  transfer_buffers_.reserve(GetAmberBuffers().size());
+  descriptor_offsets_.reserve(GetAmberBuffers().size());
+  descriptor_ranges_.reserve(GetAmberBuffers().size());
 
   for (const auto& amber_buffer : GetAmberBuffers()) {
     if (amber_buffer->ValuePtr()->empty())
       continue;
 
+    // Check if the transfer buffer is already created.
+    if (transfer_buffers_.count(amber_buffer) > 0)
+      continue;
+
     uint32_t size_in_bytes =
         amber_buffer ? static_cast<uint32_t>(amber_buffer->ValuePtr()->size())
                      : 0;
-    transfer_buffers_.emplace_back(MakeUnique<TransferBuffer>(
-        device_, size_in_bytes, amber_buffer->GetFormat()));
+
+    auto transfer_buffer = MakeUnique<TransferBuffer>(
+        device_, size_in_bytes, amber_buffer->GetFormat());
```

Comment:
The ternary expression above suggests that `amber_buffer` can be nullptr. But then we use it here anyway.  You haven't actually changed this. But it seems odd. Perhaps we should just remove the ternary expression?

Reply:
Yeah, that seems odd. `amber_buffer` is actually used also before the ternary expression:
```
if (amber_buffer->ValuePtr()->empty())
      continue;
```
I removed the ternary expression.